File: materiały/tsp_pd/tsp_data.py
#!/usr/bin/python3

def tsp_rand(size,seed):
    print("data:",size)
    for a in range(size):
        for b in range(size):
            seed = (seed*69069+1)&0xFFFFFFFF
            d = (seed%99+1)*(a!=b)
            print('{:2d}'.format(d),end=" " )
        print()
    print()    

for n in range(10,21):
    tsp_rand(n,n)

File: materiały/tsp_pd/tsp_data.cpp
#include <iostream>

void tsp_rand(int size, unsigned int seed)
{ 
  std::cout<<"data: "<<size<<"\n";  
  for(int a=0; a<size; a++){
    for(int b=0; b<size; b++){
      seed=(seed*69069+1)&0xFFFFFFFF;
      int d=(seed%99+1)*(a!=b);
      std::cout.width(2);
      std::cout<<d<<" ";}
    std::cout<<"\n";}
  std::cout<<std::endl;
}

int main()
{
  for(int i=10;i<21;i++) 
    tsp_rand(i,i);
  return 0;
}
